### PR TITLE
Ext_sequencing scenario should check agent log for extension enable order

### PR DESCRIPTION
**Pull request by maddieford**
## Description
Ext sequencing scenario should check the agent log to determine which order the extensions were enabled, instead of checking extension status file. Checking extension status file is not a good method of checking enable order, because extensions can return success without touching its status file.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests_e2e/tests/ext_sequencing/ext_sequencing.py [hunk 1]
```diff
@@ -95,10 +96,8 @@ def _get_sorted_extension_names(extensions: List[VirtualMachineScaleSetVMExtensi
         for ext in extensions:
             # Only check extensions which succeeded provisioning
             if "succeeded" in ext.statuses_summary[0].code:
-                enabled_time = ssh_client.run_command(f"ext_sequencing-get_ext_enable_time.py --ext '{extension_full_names[ext.name]}'", use_sudo=True)
-                formatted_time = datetime.strptime(enabled_time.strip(), u'%Y-%m-%dT%H:%M:%SZ')
-                if formatted_time < test_case_start:
```
Comment (maddieford) [inline tests_e2e/tests/ext_sequencing/ext_sequencing.py:100]: This check is no longer needed. ext_sequencing-get_ext_enable_time.py now only checks for logs after this timestamp

File: tests_e2e/tests/ext_sequencing/ext_sequencing.py [hunk 2]
```diff
@@ -184,7 +183,7 @@ def run(self):
         }
 
         for case in self._test_cases:
-            test_case_start = datetime.now()
+            test_case_start = random.choice(list(ssh_clients.values())).run_command("date '+%Y-%m-%d %T'").rstrip()
```
Comment (maddieford) [inline tests_e2e/tests/ext_sequencing/ext_sequencing.py:186]: Issues with time on orchestrator VM being different than the time on test ubuntu vms, as seen in other scenarios.

Using one of the VMSS instances to get the time on the test VMs

File: tests_e2e/tests/scripts/ext_sequencing-get_ext_enable_time.py [hunk 3]
```diff
@@ -20,68 +20,42 @@
...
+    after_time = datetime.strptime(args.after_time, u'%Y-%m-%d %H:%M:%S')
+    # Agent logs for extension enable: 2024-02-09T09:29:08.943529Z INFO ExtHandler [Microsoft.Azure.Extensions.CustomScript-2.1.10] Enable extension: [bin/custom-script-shim enable]
+    enable_log_regex = r"\[{0}-[.\d]+\] Enable extension: .*".format(args.ext)
+
+    agent_log = AgentLog()
+    try:
+        for agent_record in agent_log.read():
+            if agent_record.timestamp >= after_time:
+                # The agent_record prefix for enable logs is the extension name, for example: [Microsoft.Azure.Extensions.CustomScript-2.1.10]
+                if agent_record.prefix:
```
Comment (narrieta) [inline tests_e2e/tests/scripts/ext_sequencing-get_ext_enable_time.py:49]: is this testing "is not None"? can we make it explicit?

**Review by narrieta — DISMISSED** [review]

File: tests_e2e/tests/scripts/ext_sequencing-get_ext_enable_time.py (hunk 3 above)
Comment (maddieford) [reply tests_e2e/tests/scripts/ext_sequencing-get_ext_enable_time.py]: Updated

**Review by nagworld9 — APPROVED** [review]

**Pull request merged**